crates/primitives-traits/src/receipt.rs

```diff
@@ -0,0 +1,29 @@
+//! Receipt abstraction
+
+use alloc::fmt;
+
+use alloy_consensus::TxReceipt;
+use reth_codecs::Compact;
+use serde::{Deserialize, Serialize};
+
+/// Helper trait that unifies all behaviour required by receipt to support full node operations.
+pub trait FullReceipt: Receipt + Compact {}
+
+impl<T> FullReceipt for T where T: Receipt + Compact {}
```

Comment:
we can do naming bikeshed later but I like this.

do we have `alloy::Txreceipt` support for our receipt types? is this even possible rn?